SU2_CFD/include/numerics/turbulent/turb_sources.hpp

```diff
@@ -152,7 +155,43 @@ class CSourceBase_TurbSA : public CNumerics {
     Jacobian_i[0] = 0.0;
 
     /*--- Evaluate Omega with a rotational correction term. ---*/
-
+    // pgomega by Xiao He 2022
+      if (options.pgomga) {
+        
+        /*!\brief  PG-OMEGA*/
+        su2double dpds = 0, dpomg = 0, omg = 0, velocityMag = 0;
+        su2double Velocity_Rel[3] = {0.0, 0.0, 0.0};
+        su2double Vorticity_Rel[3] = {0.0, 0.0, 0.0};
+        su2double RotationalVelocity[3] = {0.0, 0.0, 0.0};
+        su2double RotationVelocityCrossR[3] = {0.0, 0.0, 0.0};
+        for (int iDim = 0; iDim < nDim; iDim++) {
+          RotationalVelocity[iDim] = config->GetRotation_Rate(iDim) / config->GetOmega_Ref();
+        }
+        
+        GeometryToolbox::CrossProduct( RotationalVelocity,Coord_i, RotationVelocityCrossR);
+        
+        for (int iDim = 0; iDim < nDim; iDim++) {
+          Velocity_Rel[iDim] = V_i[idx.Velocity() + iDim] - RotationVelocityCrossR[iDim];
+          Vorticity_Rel[iDim] = Vorticity_i[iDim] - RotationalVelocity[iDim];
+         
+        }
```

Comment:
If this model only works for 3D flows (I assume you only use it in these cases) you should not allow users to select it with a 2D calculation